### PR TITLE
feat(webchannels): Add initial support for `fxa_status` webchannel event

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We need to be able to send and recieve the `fxa_status` webchannel event

## This pull request

- Adds support for this message

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-3362

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-settings/src/lib/firefox.ts [hunk 1]
```diff
@@ -31,6 +32,28 @@ interface ProfileMetricsEnabled {
 type Profile = ProfileUid | ProfileMetricsEnabled;
 type FirefoxEvent = CustomEvent<FirefoxMessage | string>;
 
+// This is defined in the Firefox source code:
```
Comment (dschom) [inline packages/fxa-settings/src/lib/firefox.ts:35]: nit: maybe use jsdoc style?

File: packages/fxa-settings/src/lib/firefox.ts [hunk 2]
```diff
@@ -31,6 +32,28 @@ interface ProfileMetricsEnabled {
...
+// https://searchfox.org/mozilla-central/source/services/fxaccounts/tests/xpcshell/test_web_channel.js#348
```
Comment (dschom) [inline packages/fxa-settings/src/lib/firefox.ts:36]: 👍

File: packages/fxa-settings/src/models/AppContext.ts [hunk 3]
```diff
@@ -131,6 +136,8 @@ export function initializeAppContext() {
     relierFactory,
     navigatorLanguages: navigator.languages || ['en'],
   };
+  
+  firefox.send(FirefoxCommand.FxAStatus, {});
```
Comment (dschom) [inline packages/fxa-settings/src/models/AppContext.ts:140]: Do we want to use firefox.fxaStatus(...) instead?

**Review by dschom — APPROVED** [review]
Other than the call direct call to firefox.send, LGTM. 👍

File: packages/functional-tests/pages/resetPassword.ts [hunk 4]
```diff
@@ -59,7 +59,7 @@ export class ResetPasswordPage extends BaseLayout {
   }
 
   async fillOutResetPassword(email) {
-    await this.page.fill(selectors.EMAIL, email);
+    await this.page.locator(selectors.EMAIL).fill(email);
```
Comment (vbudhram) [inline packages/functional-tests/pages/resetPassword.ts:62]: Unrelated, but this fixes the flaky test in the old password reset flows.

**Pull request merged**